runtime/lua/vim/secure.lua

```diff
@@ -86,21 +109,79 @@ function M.read(path)
     trust[fullpath] = hash
   end
 
-  do
-    local f, err = io.open(vim.fn.stdpath('state') .. '/trust', 'w')
-    if not f then
-      error(err)
+  write_trust(trust)
+
+  return contents
+end
+
+--- Update the trust status of file at {path} in the trust database at
+--- $XDG_STATE_HOME/nvim/trust.
```

Comment:
```suggestion
--- Manage the trust database.
---
--- The trust database is located at |$XDG_STATE_HOME|/nvim/trust.
```

Reply:
Done in d3d5654ca469c73ed196ac902a9e1eaadf4d0ec1